seq/src/main.rs

```diff
@@ -0,0 +1,125 @@
+use std::process;
+
+use clap::{load_yaml, App, AppSettings};
+
+#[derive(Clone,Debug)]
+pub struct Seq {
+    current: f64,
+    stop: f64,
+    step: f64,
+}
+
+impl Iterator for Seq {
+    type Item = f64;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.step.is_sign_positive() && self.current <= self.stop {
+            let result = self.current.clone();
+
+            self.current += self.step;
+
+            Some(result)
+        } else if self.step.is_sign_negative() && self.current >= self.stop
```

Comment:
The logic of the inside this `else if` is the same as the first `if` logic, it's that right? If so, you could just use logical OR (`||`) on the first if and remove this `else if`

Reply:
Yeah, I refactored this to be one if condition since they are the same.